### DISCUSSION TITLE
Discord OAuth: "userinfo_endpoint must be configured on the issue

### DISCUSSION THREAD

**Discussion by Kunsi** [immich-app/immich#14452] [Feature Request]
### The bug

We're using Discord to provide SSO for our users. When configuring immich to use Discord as OAuth provider, the login flow works, however the OAuth login does not succeed. The log says `Unknown error: TypeError: userinfo_endpoint must be configured on the issuer`

### The OS that Immich Server is running on

debian bookworm

### Version of Immich Server

v1.118.2

### Version of Immich Mobile App

not used

### Platform with the issue

- [X] Server
- [X] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
# Launching directly using `docker run`
docker run -d  \
       --name "immich" \
        --env "PUID=$PUID" \
        --env "PGID=$PGID" \
        --env "TZ=UTC" \
        --env "DB_DATABASE_NAME=immich" \
        --env "DB_HOSTNAME=host.docker.internal" \
        --env "DB_PASSWORD=redacted" \
        --env "DB_USERNAME=immich" \
        --env "REDIS_HOSTNAME=host.docker.internal" \
        --network host \
        --volume "/var/opt/docker-engine/immich/config:/config" \
        --volume "/var/opt/docker-engine/immich/libraries:/libraries" \
        --volume "/var/opt/docker-engine/immich/photos:/photos" \
        --restart unless-stopped \
        "ghcr.io/imagegenius/immich:latest"
```

### Your .env content

```Shell
# not used, see above
```

### Reproduction steps

1. Configure Discord OAuth 
2. Try to link user to OAuth account
3. Get error message

### Relevant log output

```shell
[Nest] 179  - 12/02/2024, 7:04:26 PM   ERROR [Api:ErrorInterceptor~0ynllamg] Unknown error: TypeError: userinfo_endpoint must be configured on the issuer
```

### Additional information

Discord does not provide a userinfo endpoint: https://discord.com/.well-known/openid-configuration

This was also already solved in another software we're using (but issue was reported by someone else): https://github.com/plankanban/planka/discussions/875

This issue also occurs when using the docker-compose file in this repository, however due to circumstances beyond my control we cannot use that in production. I verified this in our development setup though.

**Comment by mmomjian** [comment]
Thanks for the feature request. Immich currently supports OIDC, not all OAuth, so Discord wouldn't be expected to work.

**Comment by bo0tzz** [comment]
As a workaround you could consider using something like https://dexidp.io/ as an intermediate layer. That's also nice if you're doing IaC for a bunch of different apps as then you don't need to configure a bunch of different oauth clients with Discord.
